src/libAtomVM/nifs.c

```diff
@@ -3651,6 +3658,90 @@ static term nif_erlang_make_fun_3(Context *ctx, int argc, term argv[])
     return term_make_function_reference(module_term, function_term, arity_term, &ctx->heap);
 }
 
+static term nif_erlang_fun_info_2(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+    term fun = argv[0];
+    VALIDATE_VALUE(fun, term_is_fun);
+    term key = argv[1];
+    VALIDATE_VALUE(key, term_is_atom);
+
+    term value;
+    switch (key) {
+        case MODULE_ATOM: {
+            const term *boxed_value = term_to_const_term_ptr(fun);
+            if (term_is_external_fun(fun)) {
+                term module_atom = boxed_value[1];
```

Comment:
I suggest adding some inline functions to `term.h` (or use existing ones) in order to access fun inner structure, so we don't leak outside of `term.h` the fun layout.

Reply:
Done, I used single function for that with null checks for arguments. If this adds too much branching, I can split them into three functions.
Also passing entire context, not sure if it should take global and heap separately instead.